pkg/controller/factory/base_controller.go

```diff
@@ -77,11 +78,14 @@ func (c *baseController) Sync(ctx context.Context, syncCtx SyncContext) error {
 	return c.sync(ctx, syncCtx)
 }
 
+// QueueKey return queue key for given name.
+func QueueKey(name string) string {
+	return strings.ToLower(name) + "Key"
```

Comment:
this looks weird.  Why do the other queuing mechanisms not require this?

Reply:
we usually make `const key` and use it... in factory, we have to construct it... I will need this to make resourcesynccontroller factory based as it need to access the queue from SyncSecret/ConfigMap that we export from controller...

Reply:
> we usually make `const key` and use it... in factory, we have to construct it... I will need this to make resourcesynccontroller factory based as it need to access the queue from SyncSecret/ConfigMap that we export from controller...

the issue isn't the exposure.  The issue is that this should be re-used everywhere, right?